Sources/Sentry/SentryScope.m

```diff
@@ -123,8 +123,20 @@ - (void)addBreadcrumb:(SentryBreadcrumb *)crumb
     if (self.maxBreadcrumbs < 1) {
         return;
     }
-    [SentryLog logWithMessage:[NSString stringWithFormat:@"Add breadcrumb: %@", crumb]
-                     andLevel:kSentryLevelDebug];
+
+    // since we will create a breadcrumb for each log statement, we won't log breadcrumbs with
+    // SentryLog or else we'll end up in an infinite loop. use printf instead so we can still see
+    // this in the console.
+    NSString *message = [NSString stringWithFormat:@"Add breadcrumb: %@", crumb];
+#if !defined(TESTCI) && !defined(TEST)
+    [SentryLog logWithMessage:message andLevel:kSentryLevelDebug];
+#else
+    static NSISO8601DateFormatter *df;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ df = [[NSISO8601DateFormatter alloc] init]; });
+    printf("%s: %s\n", [df stringFromDate:[NSDate date]].UTF8String, message.UTF8String);
+#endif
```

Comment:
`m`: I think it would be good to find a way that we don't need that here. What about keeping the scope as it is and let the `SentryBreadcrumbTestLogOutput` take care of breaking the endless loop? `SentryBreadcrumbTestLogOutput` could just not add a breadcrumb if the message starts with `Add breadcrumb: `. We could define a constant for `Add breadcrumb: `, which both `SentryScope` and `SentryBreadcrumbTestLogOutput` use to ensure it doesn't break when changing the log message. My suggested solution is also not the best, but I like it more than polluting the scope with test log messages.